dotnet/samples/03.adaptiveCards.a.typeAheadBot/Program.cs

```diff
@@ -27,17 +29,41 @@
 builder.Services.AddSingleton<IBotFrameworkHttpAdapter>(sp => sp.GetService<CloudAdapter>()!);
 builder.Services.AddSingleton<BotAdapter>(sp => sp.GetService<CloudAdapter>()!);
 
+// Create singleton instances for bot application
 builder.Services.AddSingleton<IStorage, MemoryStorage>();
+builder.Services.AddSingleton<ActivityHandlers>();
 
-// Create the Application.
-builder.Services.AddTransient<ApplicationOptions<TurnState, TurnStateManager>>(sp =>
+// Create the bot as a transient. In this case the ASP Controller is expecting an IBot.
+builder.Services.AddTransient<IBot>(sp =>
 {
-    return new ApplicationOptions<TurnState, TurnStateManager>()
+    IStorage storage = sp.GetService<IStorage>()!;
+    ApplicationOptions<TurnState, TurnStateManager> applicationOptions = new()
     {
-        Storage = sp.GetService<IStorage>()
+        Storage = storage,
     };
+
+    Application<TurnState, TurnStateManager> app = new(applicationOptions);
+
+    ActivityHandlers activityHandlers = sp.GetService<ActivityHandlers>()!;
+
+    // Listen for new members to join the conversation
+    app.OnConversationUpdate("membersAdded", activityHandlers.MembersAddedHandler);
+
+    // Listen for messages that trigger returning an adaptive card
+    app.OnMessage(new Regex(@"static", RegexOptions.IgnoreCase), activityHandlers.StaticMessageHandler);
+    app.OnMessage(new Regex(@"dynamic", RegexOptions.IgnoreCase), activityHandlers.DynamicMessageHandler);
+
+    // Listen for query from dynamic search card
+    app.AdaptiveCards.OnSearch("nugetpackages", activityHandlers.SearchHandler);
+    // Listen for submit buttons
+    app.AdaptiveCards.OnActionSubmit("StaticSubmit", activityHandlers.StaticSubmitHandler);
+    app.AdaptiveCards.OnActionSubmit("DynamicSubmit", activityHandlers.DynamicSubmitHandler);
+
+    // Listen for ANY message to be received. MUST BE AFTER ANY OTHER HANDLERS
+    app.OnActivity(ActivityTypes.Message, activityHandlers.MessageHandler);
```

Comment:
It makes sense why `ActivityHandler` is instantiated as a singleton and in a separate class. It allows developers to do heavy set up (like loading from a file) once and use it in the activity handlers. 

Though I was wondering if it's possible to keep the same builder service set up but have it so that

 ```c# 
 app.OnActivity(ActivityTypes.Message, activityHandler.MessageHandler)
 ``` 
 is instead 
 ```c#
 app.OnActivity(ActivityTypes.Message, () => ...)
 ``` 
 
 Using static method placeholders instead of the actual implementation - I think - takes away from the seamless readability of the fluent design.

Reply:
Keeping the above comment in mind, for separation of concerns sakes, does it makes sense to move the `app.xyx()` handler registration to a separate class?

Reply:
Good suggestion. The handlers body may be very long, then `app.OnActivity(ActivityTypes.Message, () => ...)` style will look messy. This is another reason we put all handlers together in a separate class. But agree with you that it takes away from the seamless readability of the fluent design.
Do you mind create an issue for that? Then we can get the PR merged first and fix the issue later.

Reply:
What do you mean of moving the `app.xyx()` handler registration to a separate class?

Reply:
I mean instead of registering handlers (i.e `app.message()` ...etc) in `Program.cs`, can we do it in a separate class instead? This will make the `app.OnActivity(ActivityTypes.Message, () => ...)` less messy.

Reply:
> Good suggestion. The handlers body may be very long, then `app.OnActivity(ActivityTypes.Message, () => ...)` style will look messy. This is another reason we put all handlers together in a separate class. But agree with you that it takes away from the seamless readability of the fluent design. Do you mind create an issue for that? Then we can get the PR merged first and fix the issue later.

Great. Here's the ticket: https://github.com/orgs/microsoft/projects/719?pane=issue&itemId=38284723

Reply:
> I mean instead of registering handlers (i.e app.message() ...etc) in Program.cs, can we do it in a separate class instead? This will make the app.OnActivity(ActivityTypes.Message, () => ...) less messy.

I see, good idea. Thanks for explanation.